cameraview/src/main/java/com/otaliastudios/cameraview/SnapshotPictureRecorder.java

```diff
@@ -54,18 +65,35 @@ private void takeGl(@NonNull final GlCameraPreview preview) {
         preview.addRendererFrameCallback(new GlCameraPreview.RendererFrameCallback() {
 
             int mTextureId;
+            int mOverlayTextureId = 0;
             SurfaceTexture mSurfaceTexture;
+            SurfaceTexture mOverlaySurfaceTexture;
             float[] mTransform;
+            float[] mOverlayTransform;
+            EglViewport viewport;
 
             @RendererThread
             public void onRendererTextureCreated(int textureId) {
                 mTextureId = textureId;
+                viewport = new EglViewport();
+                if (mWithOverlay) {
+                    mOverlayTextureId = viewport.createTexture();
+                }
                 mSurfaceTexture = new SurfaceTexture(mTextureId, true);
+                if (mOverlayTextureId != 0) {
```

Comment:
Can you use if (mWithOverlay) instead?

Reply:
Hey, the value mWithOverlay used to be `false` only if `disableOverlayFor` took the `PICTURE` value. Now that the `disableOverlayFor` is gone, `mWithOverlay` will always be true. I added a `TODO` comment since I guess we'll fix this behavior with the API discussed in the issue.